corehq/apps/userreports/tests/test_choice_provider.py

```diff
@@ -0,0 +1,305 @@
+from abc import ABCMeta, abstractmethod
+from django.test import SimpleTestCase, TestCase
+import mock
+from corehq.apps.commtrack.tests.util import bootstrap_location_types, make_loc
+from corehq.apps.es.fake.groups_fake import GroupESFake
+from corehq.apps.es.fake.users_fake import UserESFake
+from corehq.apps.groups.models import Group
+from corehq.apps.locations.tests import delete_all_locations
+from corehq.apps.reports_core.filters import Choice
+from corehq.apps.userreports.models import ReportConfiguration
+from corehq.apps.userreports.reports.filters.choice_providers import ChoiceProvider, \
+    ChoiceQueryContext, LocationChoiceProvider, UserChoiceProvider, GroupChoiceProvider, \
+    OwnerChoiceProvider
+from corehq.apps.users.models import CommCareUser, WebUser, DomainMembership
+from corehq.apps.users.util import normalize_username
+
+
+class SearchableChoice(Choice):
+    def __new__(cls, value, display, searchable_text=None):
+        self = super(SearchableChoice, cls).__new__(cls, value, display)
+        self.searchable_text = [text for text in searchable_text or []
+                                if text is not None]
+        return self
+
+
+class StaticChoiceProvider(ChoiceProvider):
```

Comment:
Yeah it should be a full implementation of the `ChoiceProvider` interface, so theoretically you could use it in production for a fixed set of choices or a short but slightly dynamic list of choices, though nothing uses it currenlty.